postinstall.js

```diff
@@ -71,8 +72,17 @@ function hasCache() {
 function download(cb) {
 	console.log('ngrok - downloading binary ' + cdnUrl);
 
+	const caString = execSync('npm config get ca', {stdio: ['ignore', 'pipe', 'pipe']})
```

Comment:
did you try ```NODE_OPTIONS=--use-openssl-ca ?```

or ```npm config set strict-ssl```

or ```NODE_EXTRA_CA_CERTS``` option
https://github.com/nodejs/node/issues/14705

or ```NODE_TLS_REJECT_UNAUTHORIZED=0```
https://stackoverflow.com/questions/29283040/how-to-add-custom-certificate-authority-ca-to-nodejs

Reply:
I didn't see these in the docs, but I'll admit some ignorance here about how node interacts with the request module, and so my searches were likely missing the mark.

I will check these out when I'm back in the office (I'm going in at some point today). The issue you reference doesn't seem to make it clear whether that environment variable will be respected if the ca argument isn't set.

As for setting strict-ssl to false, that does work for installing packages, but the request module doesn't care. It's also a workaround that leaves you open to MITM attacks, and even if it's unlikely, I think it's best not to require that workaround for corporate networks that are using their own CA chain.

I haven't tried ...REJECT_UNAUTHORIZED, but ditto to the above security concerns.

If the ..CA_CERTS environment variable is respected by the request module, then this is just a documentation update, and I can remove this code. I'll let you know later today.

The comments below assume that the environment variable isn't read:

I use yarn as well. It respects npmrc's settings, and as such it's the only place you need to set the ca/cafile options. Is it better to document this, or should I break this logic out into a function and test both? The logic is pretty dead simple as it stands, which is a plus. I will test the case of setting the caFile in yarnrc but not npmrc. I'm not 100% sure yarn will fully function in that scenario, which would mean we could safely ignore that case.

Also note that the second commit reverses the certificate test and prevents crashing. Worst case scenario here is that the ca array ends up undefined, in which case the request module just ignores it and runs into the same issue that I had.

Note for myself. I need to pass the resulting array instead of just sending the first result. The docs mention that a CA chain can be defined, which means it's possible to have multiple certs here.

Reply:
It would appear that none of these options work for the postinstall script (unless I write code in the postinstall script to handle them explicitly), even if ca is not specified. Reading the referenced issue more carefully, it would seem that is their intent - they want the package authors to be able to override node's default behavior. One person mentions adding a feature to node to read these in without having to read the file the way I am, but it looks like they didn't add it.